Makefile

```diff
@@ -0,0 +1,30 @@
+CC := gcc
+
+LIBS := -lstdc++ -lavcodec -lavformat -lavutil -lswresample -lswscale -lglfw -lGL
+TARGET_DIR := bin/
+
+SRC_FILES_PLAYER := main_window.cpp decoder.cpp
+SRC_FILES_TCP_PLAYER := tcp_main_window.cpp decoder.cpp tcp_client.c
+SRC_FILES_TCP_SERVER := tcp_server.c
+SRC_FILES_TRANSCODER := transcoder.c debug.c
+
+.PHONY: all
+
+all: dir player tcp_player tcp_server transcoder
+
+dir:
+	if [ ! -d "$(TARGET_DIR)" ]; then \
+		mkdir -p $(TARGET_DIR); \
+	fi
+	
```

Comment:
Can't this be automated? Rather than asking the user to do `make dir` can't we have it prepared when any of the binary is compiled?

Reply:
the user just needs to put make and everything else will be done

Reply:
okay, but consider a case if the user only wants to prepare the TCP player binary and there isn't a bin/ dir. what would be the result then? Would makefile report an error?

Reply:
because I have specified the dir target inside all

Reply:
You sure about that? would you mind rechecking the scenario I explained and get back to me?

Reply:
ohhhh I get you
will work on that soon